src/compile/data/formatparse.ts

```diff
@@ -141,12 +147,12 @@ export class ParseNode extends DataFlowNode {
    * Creates a parse node from "explicit" parse and "implicit" parse and updates ancestorParse.
    */
   private static makeWithAncestors(parent: DataFlowNode, explicit: Dict<string>, implicit: Dict<string>, ancestorParse: AncestorParse) {
-    // We should not parse what has already been parsed in a parent (explicitly or implicitly) or what has been derived (maked as "derived").
+    // We should not parse what has already been parsed in a parent (explicitly or implicitly) or what has been derived (maked as "derived") and we don't need to flatten again.
```

Comment:
this sentence is now super complicated.  I don't understand how "or" and "and" are being grouped.

Reply:
Agreed and fixed